javaagent/sdk-migration-config.yaml

```diff
@@ -0,0 +1,291 @@
+            # Configure headers. Entries have higher priority than entries from .headers_list.
+            headers: []
+            # Configure headers. Entries have lower priority than entries from .headers.
+            # The value is a list of comma separated key-value pairs matching the format of OTEL_EXPORTER_OTLP_HEADERS. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options for details.
+            headers_list: ${OTEL_EXPORTER_OTLP_TRACES_HEADERS}
+  # Configure span limits. See also attribute_limits.
+  limits:
+    # Configure max attribute value size. Overrides .attribute_limits.attribute_value_length_limit.
+    attribute_value_length_limit: ${OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT}
+    # Configure max attribute count. Overrides .attribute_limits.attribute_count_limit.
+    attribute_count_limit: ${OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT:-128}
+    # Configure max span event count.
+    event_count_limit: ${OTEL_SPAN_EVENT_COUNT_LIMIT:-128}
+    # Configure max span link count.
+    link_count_limit: ${OTEL_SPAN_LINK_COUNT_LIMIT:-128}
+    # Configure max attributes per span event.
+    event_attribute_count_limit: ${OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT:-128}
+    # Configure max attributes per span link.
+    link_attribute_count_limit: ${OTEL_LINK_ATTRIBUTE_COUNT_LIMIT:-128}
+  # Configure the sampler.
+  sampler:
+    # Configure sampler to be parent_based.
+    parent_based:
+      # Configure root sampler.
+      root:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure remote_parent_sampled sampler.
+      remote_parent_sampled:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure remote_parent_not_sampled sampler.
+      remote_parent_not_sampled:
+        # Configure sampler to be always_off.
+        always_off: {}
+      # Configure local_parent_sampled sampler.
+      local_parent_sampled:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure local_parent_not_sampled sampler.
+      local_parent_not_sampled:
+        # Configure sampler to be always_off.
+        always_off: {}
+      # Uncomment to configure the rule_based_routing sampler from https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/samplers
+      # to drop spans to spring actuator endpoints.
+      # Configure the parent_based sampler's root sampler to be rule_based_routing sampler.
+      # root:
+      #  rule_based_routing:
+      #    # Fallback to the always_on sampler if the criteria is not met.
+      #    fallback_sampler:
+      #      always_on:
+      #    # Only apply to SERVER spans.
+      #    span_kind: SERVER
+      #    rules:
+      #      # Drop spans where url.path matches the regex /actuator.* (i.e. spring boot actuator endpoints).
+      #      - action: DROP
+      #        attribute: url.path
+      #        pattern: /actuator.*
```

Comment:
Check out this fun snippet! This is the declarative config solution to https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1060